### DISCUSSION TITLE
How do I set up the latest Twitter Oauth?

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#5915] [Questions] [answered]
What is the latest way to set up Twitter Oauth?

I watched the reference video and set it up.
Unlike the video, Twitter's Developer Portal is new.
https://www.youtube.com/watch?v=z1rWsRb4PDw

`http://localhost:3000/?error=server_error&error_description=Error+getting+user+email+from+external+provider`
is returned.

Are there any incorrect settings?

![スクリーンショット 2022-03-14 19 45 27](https://user-images.githubusercontent.com/42270511/158156996-cd87c476-4011-47b2-9bf9-815c120daad9.png)

![スクリーンショット 2022-03-14 19 45 43](https://user-images.githubusercontent.com/42270511/158157199-73d7a099-0ea3-4b98-b3c0-76ab896b7223.png)

key uses Consumer Keys.
![スクリーンショット 2022-03-14 19 47 26](https://user-images.githubusercontent.com/42270511/158157278-c98c28f8-1cc3-4072-bfde-afdafe5e1a9f.png)

```js
  const handleLoginTwitter = async () => {
    await supabase.auth.signIn(
      {
        provider: "twitter",
      }
    );
  };
```

![スクリーンショット 2022-03-14 19 51 43](https://user-images.githubusercontent.com/42270511/158158010-3022ca3b-e073-4a15-983e-378562478eca.png)

**Comment by magoz** [comment]
Same issue here.
I've also tried using `Access Token and Secret`, and I'm also getting the error `http://localhost:3000/?error=server_error&error_description=Error+getting+user+email+from+external+provider`

<img width="446" alt="supabase2" src="https://user-images.githubusercontent.com/9190753/168739829-8579d0ab-ee1f-4aa9-8467-cd3904498fd1.png">

What's the right way of setting up Twitter Oauth?

**Comment by magoz** [comment, marked as answer]
If somebody else has the same problem, I solved it by applying to [Elevated](https://developer.twitter.com/en/portal/products/elevated) access.

**Reply by ru88s** [reply]
I didn't realize I had received a confirmation email from Twitter.
After Elevated was approved, I was able to log in successfully.
Thank you!
